x-pack/plugins/fleet/server/services/epm/registry/registry_url.ts

```diff
@@ -20,8 +20,8 @@ const SNAPSHOT_REGISTRY_URL_CDN = 'https://epr-snapshot.elastic.co';
 // const SNAPSHOT_REGISTRY_URL_NO_CDN = 'https://epr-snapshot.ea-web.elastic.dev';
 
 const getDefaultRegistryUrl = (): string => {
-  const branch = appContextService.getKibanaBranch();
-  if (branch === 'main') {
+  const isProduction = appContextService.getIsProductionMode();
+  if (!isProduction) {
```

Comment:
should we have both `!isProduction || branch === 'main'`, to get this working on cloud snapshot?

Reply:
just checked and looks like on 8.6 cloud snapshot we use the snapshot repository we probably want to keep that

Reply:
+1 to this - I think we need the `branch === 'main'` check here.